Clover/app/src/main/java/org/floens/chan/ui/controller/export/ImportExportSettingsController.java

```diff
@@ -0,0 +1,306 @@
+import org.floens.chan.ui.helper.RuntimePermissionsHelper;
+import org.floens.chan.ui.settings.LinkSettingView;
+import org.floens.chan.ui.settings.SettingsController;
+import org.floens.chan.ui.settings.SettingsGroup;
+import org.floens.chan.utils.AndroidUtils;
+
+import java.io.File;
+import java.util.Locale;
+
+public class ImportExportSettingsController extends SettingsController implements
+        ImportExportSettingsPresenter.ImportExportSettingsCallbacks {
+
+    @Nullable
+    private ImportExportSettingsPresenter presenter;
+
+    @Nullable
+    private OnExportSuccessCallbacks callbacks;
+
+    private LoadingViewController loadingViewController;
+    private File settingsFile = new File(ChanSettings.saveLocation.get(), ImportExportRepository.EXPORT_FILE_NAME);
+
+    public ImportExportSettingsController(Context context, @NonNull OnExportSuccessCallbacks callbacks) {
+        super(context);
+
+        this.callbacks = callbacks;
+        this.loadingViewController = new LoadingViewController(context);
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        navigation.setTitle(R.string.settings_import_export);
+
+        presenter = new ImportExportSettingsPresenter(this);
+
+        setupLayout();
+        populatePreferences();
+        buildPreferences();
+
+        showCreateCloverDirectoryDialog();
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+
+        if (presenter != null) {
+            presenter.onDestroy();
+        }
+
+        callbacks = null;
+    }
+
+    protected void setupLayout() {
+        view = inflateRes(R.layout.settings_layout);
+        content = view.findViewById(R.id.scrollview_content);
+    }
+
+    private void populatePreferences() {
+        // Import/export settings group
+        {
+            SettingsGroup group = new SettingsGroup(context.getString(R.string.import_or_export_settings));
+
+            group.add(new LinkSettingView(this,
+                    context.getString(R.string.export_settings),
+                    context.getString(R.string.export_settings_to_a_file),
+                    v -> onExportClicked()));
+
+            group.add(new LinkSettingView(this,
+                    context.getString(R.string.import_settings),
+                    context.getString(R.string.import_settings_from_a_file),
+                    v -> onImportClicked()));
+
+            groups.add(group);
+        }
+    }
+
+    private void onExportClicked() {
+        String state = Environment.getExternalStorageState();
+        if (!Environment.MEDIA_MOUNTED.equals(state)) {
+            showMessage(context.getString(R.string.error_external_storage_is_not_mounted));
+            return;
+        }
+
+        getPermissionHelper().requestPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE, granted -> {
+            if (granted) {
+                navigationController.presentController(loadingViewController);
```

Comment:
Probably should move this inside the check below because if the presenter is not set then we will show the loading view but won't start the exporing so it will just be shown forever.